### PR TITLE
Add description attribute to Attachment

### LINKED ISSUE #483: Add description to attachments
The description property is missing in attachments (https://docs.pycord.dev/en/master/api.html?highlight=attachment#discord.Attachment).

It is supported and used for screenreader / alt text.
See https://discord.com/developers/docs/resources/channel#attachment-object-attachment-structure

The field is a optional string.

### PR DESCRIPTION
**Pull request by plun1331**
## Summary

Adds the `description` attribute to `Attachment`

*This can also be added to `File`, but I tried and couldn't get it to work*

Resolves #483 (I guess)

## Checklist

- [x] If code changes were made then they have been tested.
    - [x] I have updated the documentation to reflect the changes.
- [ ] If `type: ignore` comments were used, a comment is also left explaining why
- [ ] This PR fixes an issue.
- [x] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, typehinting, examples, ...)

### CODE REVIEW

**Comment by Lulalaby** [comment]
Yep that's correct ❤️

**Comment by Lulalaby** [comment]
I'm a bit tired rn and on phone. Did you added description to adding files too?
If you send a message with a file you can add the description. The topic is a bit difficult.

**Comment by Lulalaby** [comment]
If not, we can merge this so bots can already access the property.

**Comment by plun1331** [comment]
> I'm a bit tired rn and on phone. Did you added description to adding files too? If you send a message with a file you can add the description. The topic is a bit difficult.

i spent a few hours trying but discord seems to dislike me
but no, i didn't

**Review by Lulalaby — APPROVED** [review]

**Comment by Lulalaby** [comment]
> > I'm a bit tired rn and on phone. Did you added description to adding files too? If you send a message with a file you can add the description. The topic is a bit difficult.
> 
> i spent a few hours trying but discord seems to dislike me
> but no, i didn't

Ok I'm gonna make a new issue for that with additional informations.

**Comment by Lulalaby** [comment]
@plun1331 if you're interested how it would work, you could take a look here: https://github.com/Aiko-IT-Systems/DisCatSharp/pull/81

Not many libs has it implemented yet. Discord sneaked it into the docs without any notice :D

**Comment by plun1331** [comment]
> @plun1331 if you're interested how it would work, you could take a look here: [Aiko-IT-Systems/DisCatSharp#81](https://github.com/Aiko-IT-Systems/DisCatSharp/pull/81)
> 
> Not many libs has it implemented yet. Discord sneaked it into the docs without any notice :D

I tried to implement it in a fork but discord kept sending me 400s and I couldn't figure out why

**Comment by Lulalaby** [comment]
```json
{
   "code":50035,
   "errors":{
      "attachments":{
         "0":{
            "_errors":[
               {
                  "code":"ATTACHMENT_NOT_FOUND",
                  "message":"Attachment data not found"
               }
            ]
         }
      },
      "message":"Invalid Form Body"
   }
}
```

Is that the error you get

**Comment by Lulalaby** [comment]
A correct payload_json would be:
```json
{
   "tts":false,
   "allowed_mentions":{
      "parse":[
         "everyone",
         "users",
         "roles"
      ]
   },
   "attachments":[
      {
         "filename":"cute.jpg",
         "description":"Cute anime girl UwU",
         "size":0,
         "id":0
      }
   ]
}
```

And the files name in the file form-data has to be `files[0]` in this case.
`files[n]` must be mapped to attachments: [ { "id": n }]

**Comment by plun1331** [comment]
> A correct payload_json would be:
> 
> ```json
> {
>    "tts":false,
>    "allowed_mentions":{
>       "parse":[
>          "everyone",
>          "users",
>          "roles"
>       ]
>    },
>    "attachments":[
>       {
>          "filename":"cute.jpg",
>          "description":"Cute anime girl UwU",
>          "size":0,
>          "id":0
>       }
>    ]
> }
> ```
> 
> And the files name in the file form-data has to be `files[0]` in this case. `files[n]` must be mapped to `attachments: [{ "id": n }]`

iirc thats what i was sending, but ill check

**Comment by Lulalaby** [comment]
Check if the file names are really files[n] and not encoded

**Comment by plun1331** [comment]
threw in a print and it's claiming this is being sent
```
[
  (<MultiDict('name': 'payload_json')>, 
  {}, 
  '{"tts":null,"content":"a","message_reference":{"message_id":913294318368862249,"channel_id":889181570932551692,"guild_id":889181570932551690},"attachments":[{"id":0,"description":"Profile banner","filename":"banner.jpg"}]}'), 
  (<MultiDict('name': 'files[0]', 'filename': 'banner.jpg')>, 
  {'Content-Type': 'application/octet-stream'}, 
  <_io.BufferedReader name='banner.jpg'>)
]
```

**Comment by Lulalaby** [comment]
Wrong content type

**Comment by plun1331** [comment]
> Wrong content type

thats what the lib set it to originally, what should it be set to

**Comment by Lulalaby** [comment]
application/form-data

**Comment by Lulalaby** [comment]
Wait

**Comment by Lulalaby** [comment]
No the media content type is wrong. It should be image/jpeg

**Comment by plun1331** [comment]
> application/form-data

for the file?
also, could we continue this on discord

**Comment by Lulalaby** [comment]
The media content type headers are an evil topic

**Comment by Lulalaby** [comment]
> > application/form-data
> 
> for the file?
> also, could we continue this on discord

Yus let's do that

File: discord/message.py [hunk 1]
```diff
@@ -169,6 +171,7 @@ def __init__(self, *, data: AttachmentPayload, state: ConnectionState):
         self._http = state.http
         self.content_type: Optional[str] = data.get('content_type')
         self.ephemeral: bool = data.get('ephemeral', False)
+        self.description: str = data.get('description')
```
Comment (nerdguyahmad) [inline discord/message.py:174]:
```suggestion
        self.description: Optional[str] = data.get('description')
```

**Review by nerdguyahmad — CHANGES_REQUESTED** [review]

File: discord/message.py [hunk 2]
```diff
@@ -152,11 +152,13 @@ class Attachment(Hashable):
         The attachment's `media type <https://en.wikipedia.org/wiki/Media_type>`_
     ephemeral: :class:`bool`
         Whether the attachment is ephemeral or not.
+    description: Optional[:class:`str`]
+        The attachment's description. 
 
         .. versionadded:: 1.7
```
Comment (nerdguyahmad) [inline discord/message.py:158]:
```suggestion
        .. versionadded:: 2.0
```

**Review by nerdguyahmad — CHANGES_REQUESTED** [review]

**Comment by plun1331** [comment]
Continued in #509

**Pull request closed without merging**